src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -252,15 +252,29 @@ public final SAMSequenceDictionary getReferenceDictionary() {
 
     /**
      * Returns the "best available" sequence dictionary. This will be the reference sequence dictionary if
-     * there is a reference, otherwise it will be the sequence dictionary constructed from the reads if
-     * there are reads, otherwise it will be null.
+     * there is a reference.
+     * Otherwise, if there are reads, it will be the sequence dictionary constructed from the reads.
+     * Otherwise, if there are features, it will be the first of the sequence dictionaries constructed from
+     * the VCF headers of variant Feature inputs.
+     * Otherwise, the result will be null.
      *
      * TODO: check interval file(s) as well for a sequence dictionary
      *
      * @return best available sequence dictionary given our inputs
      */
-    public final SAMSequenceDictionary getBestAvailableSequenceDictionary() {
-        return reference != null ? reference.getSequenceDictionary() : (reads != null ? reads.getSequenceDictionary() : null);
+    public SAMSequenceDictionary getBestAvailableSequenceDictionary() {
+        if (hasReference()){
+            return reference.getSequenceDictionary();
+        } else if (hasReads()){
+            return reads.getSequenceDictionary();
+        } else if (hasFeatures()){
+            final List<SAMSequenceDictionary> dictionaries = features.getVariantSequenceDictionaries();
+            //If there is just one, it clearly is the best. Otherwise, noone is best.
+            if (dictionaries.size() == 1){
+                return dictionaries.get(0);
```

Comment:
Perhaps `VariantWalker` should override `getBestAvailableSequenceDictionary()` to prioritize the sequence dictionary from the driving variants over other variant sequence dictionaries? (or create a `getBestAvailableVariantDictionary()` called from this method and overridden by `VariantWalker`)